compass-js-frontend/src/main/webapp/ui/page/editor.js

```diff
@@ -17,20 +18,40 @@ var EditorPage = BasePage.extend({
 	pageTitle: 'Editor',
 	template: template,
 	scenario: undefined,
+	root: undefined,
 	initialize: function (scenarioId, options) {
 		//The router gives as a string, but the model wants a number...
 		var idAsNumber = parseInt(scenarioId);
 		this.scenario = new Scenario({id: idAsNumber});
-		this.scenario.fetch({
-			success: this.initUI.bind(this)
-		});
+		this.fetchData();
 	},
 	render: function () {
 		this.renderWithTemplate();
 		return this;
 	},
-	initUI: function () {
+	initUI: function (rootNode) {
```

Comment:
this is still a bit of a misnomer, since it does not actually initialize UI at all.
Also it might get called before `render()` is done under odd conditions, so it's not exactly safe to even initialize UI here.